src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -87,14 +86,13 @@ public void setupSchema(List<ColumnMetadata> columns) {
     int id = 1;
     for (ColumnMetadata column : columns) {
       validateColumnCollation(column);
-      ParquetTypeGenerator.ParquetTypeInfo typeInfo =
-          ParquetTypeGenerator.generateColumnParquetTypeInfo(column, id);
+      ParquetTypeInfo typeInfo = ParquetTypeGenerator.generateColumnParquetTypeInfo(column, id);
       parquetTypes.add(typeInfo.getParquetType());
       this.metadata.putAll(typeInfo.getMetadata());
       int columnIndex = parquetTypes.size() - 1;
       fieldIndex.put(
           column.getInternalName(),
-          new ParquetColumn(column, columnIndex, typeInfo.getPrimitiveTypeName()));
+          new ParquetColumn(column, columnIndex, typeInfo.getParquetType()));
```

Comment:
where do we validate that only primitive typed typeInfos are allowed for FDN tables?

Reply:
The FDN type -> parquetType function `generateColumnParquetTypeInfo` only returns primitive type. Should we add assertion here?